go/worker/keymanager/worker.go

```diff
@@ -178,29 +184,20 @@ func (w *Worker) callLocal(ctx context.Context, data []byte) ([]byte, error) {
 }
 
 func (w *Worker) updateStatus(status *api.Status, startedEvent *host.StartedEvent) error {
-	var initOk bool
 	defer func() {
-		if !initOk {
-			// TODO: once #2130 is done and keymanager reports as Ready only
-			// after initialization succeeds, change this to always pre-register
-			// on the initial updateStatus call (and not only after the first
-			// initialization fails as it is done currently).
-			// This is likely a new key manager that needs to replicate.
-			// Send a node registration anyway, so that other nodes know
-			// to update their access control.
-			w.roleProvider.SetAvailable(func(n *node.Node) error {
-				rt := n.AddOrUpdateRuntime(w.runtime.ID())
-				rt.Version = startedEvent.Version
-				rt.ExtraInfo = nil
-				rt.Capabilities.TEE = startedEvent.CapabilityTEE
-				return nil
-			})
-
-			// If initialization failed setup a retry ticker.
-			if w.initTicker == nil {
-				w.initTicker = backoff.NewTicker(backoff.NewExponentialBackOff())
-				w.initTickerCh = w.initTicker.C
-			}
+		<-w.Initialized()
```

Comment:
Huh, I don't get this? This seems to defeat the point of this handler (pre-registration on failures) and may lead to deadlocks (`updateStatus` must be called to close `initCh` so if it failed then this deferred handler will block forever as it will block the event loop from propagating more status updates). Also wasn't the point mentioned in the removed comment that this should be changed to always pre-register?

cc @ptrus @Yawning

Reply:
This change breaks the replication code in various ways yes.  Key managers need to either register with a populated `ExtraInfo` field (ready to service requests), or with an empty one (replication needed).

Trampling over the field like in this PR will ensure that newly added replicas will never be used, till the nodes are restarted (I think).

Reply:
`SetAvailable()` is now first called for pre-registration with empty extra-info (to enable replication) and then again after successful initialization with populated extra-info.